app/uk/gov/hmrc/trackingconsentfrontend/assets/test/domain/userPreference.spec.ts

```diff
@@ -88,6 +129,17 @@ describe('User Preference Factory', () => {
         settings: true
       })
     })
+
+    it('should not return each category when AcceptAll is non-boolean but truthy', () => {
+      setConsentCookie({
+        preferences: {
+          acceptAll: 'abc'
+        }
+      })
+
+      expect(userPreferenceFactory().getPreferences()).toEqual({})
```

Comment:
Should this be returning `undefined`?  Here's what I'd expect:

`{ usage: true }` -> `{ usage: true }`
`{ usage: true, settings: 'abc' }` -> `{ usage: true }` (unacceptable value ignored)
`{ usage: true, someNewThing: true }` -> `{ usage: true }` (unknown key ignored)
`{ usage: 'abc', someNewThing: true }` -> `undefined` (unacceptable value ignored, unknown key ignored, no preferences to return)

Essentially the approach I'm suggesting is to treat unreadable answers as though they didn't exist, if after that nothing exists then `undefined` is a good thing to return.